scripts/reboot_helper.py

```diff
@@ -0,0 +1,224 @@
+        if platform_chassis is None:
+            log.log_error("Platform chassis is not loaded")
+            return False
+        return True
+    except Exception as e:
+        log.log_error(f"Failed to instantiate Chassis: {e}")
+        return False
+
+
+def load_platform_json(platform):
+    """Load and return the content of platform.json."""
+    platform_json_path = os.path.join("/usr/share/sonic/device", platform, "platform.json")
+    try:
+        with open(platform_json_path, 'r') as platform_json:
+            return json.load(platform_json)
+    except FileNotFoundError:
+        log.log_error("platform.json not found")
+    except json.JSONDecodeError:
+        log.log_error("Failed to parse platform.json")
+    return None
+
+
+def get_all_dpus():
+    """
+    Retrieve a list of all DPUs (Data Processing Units) in the system.
+    This function checks if the platform is a smartswitch and then loads the platform.json
+    file to extract the DPUs dictionary. It converts the DPU names to uppercase and returns
+    them as a list.
+
+    Returns:
+        list: A list of DPU names in uppercase.
+    """
+    if not is_smartswitch():
+        return []
+
+    platform_info = device_info.get_platform_info()
+    platform = platform_info.get('platform')
+    if not platform:
+        log.log_error("Platform does not exist in platform_info")
+        return []
+
+    config_data = load_platform_json(platform)
+    if config_data is None:
+        return []
+
+    dpus = config_data.get("DPUS", [])
+    return [dpu.upper() for dpu in dpus]
+
+
+def reboot_module(module_name):
+    """
+    Reboot the specified module by invoking the platform API.
+
+    Args:
+        module_name (str): The name of the module to reboot.
+
+    Returns:
+        bool: True if the reboot command was successfully sent, False otherwise.
+    """
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch to reboot module")
+        return False
+
+    # Attempt to reboot the module
+    if not hasattr(platform_chassis, 'reboot'):
+        log.log_error("Reboot method not found in platform chassis")
+        return False
+
+    if module_name.upper() not in get_all_dpus():
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    log.log_info("Rebooting module {}...".format(module_name))
+    try:
+        platform_chassis.reboot(module_name)
+        log.log_info("Reboot command sent for module {}".format(module_name))
+    except NotImplementedError:
+        raise NotImplementedError("Reboot not implemented on this platform: {type(e).__name__}")
+    except Exception as e:
+        log.log_error("Unexpected error occurred while rebooting module {}: {}".format(module_name, e))
+        return False
+
+    return True
+
+
+def is_dpu():
```

Comment:
Why don't we add this to the device_info.py file (in the same way as is_smartswitch(): https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-py-common/sonic_py_common/device_info.py#L579) to prevent code duplication. It will allow to reuse it in many places.

Reply:
Good point. I will do this change first and then refactor the code. Thanks!

Reply:
created a PR https://github.com/sonic-net/sonic-buildimage/pull/20724 for these smartswitch common DPU helpers.